test/zipkin/span_context_test.cc

```diff
@@ -0,0 +1,152 @@
+#include "zipkin/span_context.h"
+#include "zipkin/zipkin_core_constants.h"
+
+#include "gtest/gtest.h"
+
+namespace Zipkin {
+
+TEST(ZipkinSpanContextTest, populateFromString) {
+  SpanContext spanContext;
+
+  // Non-initialized span context
+  EXPECT_EQ(0ULL, spanContext.trace_id());
+  EXPECT_EQ("0000000000000000", spanContext.traceIdAsHexString());
+  EXPECT_EQ(0ULL, spanContext.id());
+  EXPECT_EQ("0000000000000000", spanContext.idAsHexString());
+  EXPECT_EQ(0ULL, spanContext.parent_id());
+  EXPECT_EQ("0000000000000000", spanContext.parentIdAsHexString());
+  EXPECT_FALSE(spanContext.isSetAnnotation().cr);
+  EXPECT_FALSE(spanContext.isSetAnnotation().cs);
+  EXPECT_FALSE(spanContext.isSetAnnotation().sr);
+  EXPECT_FALSE(spanContext.isSetAnnotation().ss);
+  EXPECT_EQ("0000000000000000;0000000000000000;0000000000000000", spanContext.serializeToString());
+
+  // Span context populated with trace id, id, parent id, and no annotations
```

Comment:
not sure if here or not, but should test that 128bit variant doesn't crash. All new instrumentation support 128-bit trace ids, so if you accept an inbound trace id, the choices are to truncate by dropping high bits, or better.. use all of them.